frontend/components/exercise/FillInTheBlank.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+
+interface FillInTheBlankProps {
+  id: string;
+  answer: string;
+  correctAnswer: string;
+  onAnswerChange: (answer: string) => void;
+}
+
+const FillInTheBlank: React.FC<FillInTheBlankProps> = ({
+  id,
+  answer,
+  correctAnswer,
+  onAnswerChange,
+}) => {
+  const [isActive, setIsActive] = useState(false);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onAnswerChange(e.target.value);
+  };
+
+  const handleReset = () => {
+    onAnswerChange("");
+  };
+
+  const containerWidth =
+    Math.max(correctAnswer.length, "Type here".length) * 10 + 20;
+
+  return (
+    <div
+      className="relative inline-block flex h-[32px] items-center rounded-md border border-[#EDEDED] bg-white"
+      style={{
+        width: containerWidth,
+        padding: "0px 10px 0px 12px",
+      }}
+    >
+      <input
+        type="text"
+        value={answer}
+        onChange={handleInputChange}
+        onFocus={() => setIsActive(true)}
+        onBlur={() => setIsActive(false)}
+        className={`h-full w-full font-medium outline-none text-sm ${
+          isActive
+            ? "text-[#BFBFBF] placeholder-[#BFBFBF]"
+            : "text-[#292929] placeholder-[#999999]"
+        }`}
+        placeholder="Type here"
+        style={{
+          paddingRight: "8px",
+        }}
+      />
+      {answer && (
+        <button
+          onClick={handleReset}
+          className="absolute right-[10px] top-1/2 -translate-y-1/2 transform"
+        >
+          &#10005; {/* X icon as a reset button */}
```

Comment:
use IconButton. it has hover + press states. in Figma it's linked to "button-icon-small". i didn't notice this until pretty late lol but if you see that something is a component in Figma you can find it in the components and sometimes they have hover + press states